bigquery_etl/format_sql/formatter.py

```diff
@@ -0,0 +1,189 @@
+
+    def add(self, token):
+        """Add a token to this line."""
+        self.inline_length += len(token.value)
+        self.inline_tokens.append(token)
+        self.contains_comment = self.contains_comment or isinstance(token, Comment)
+        self.can_start_inline_block = isinstance(token, OpeningBracket)
+
+    @property
+    def tokens(self):
+        """Get a list of all the tokens in this line."""
+        if self.indent_token is None:
+            return self.inline_tokens
+        else:
+            return [self.indent_token] + self.inline_tokens
+
+
+def inline_block_format(tokens, max_inline_length=100):
+    """Extend simple_format with inline blocks.
+
+    Inline a bracket block if it would be shorter than max_inline_length.
+
+    Do not inline bracket blocks that contain a comment.
+
+    Implementation assumes simple_format will put opening brackets
+    at the end of the line and closing brackets at the beginning of the line,
+    unless there is a comment between them.
+    """
+    # format tokens using simple_format, then group into lines
+    lines = [Line()]
+    for token in simple_format(tokens):
+        if token.value.startswith("\n"):
+            lines.append(Line(token))
+        else:
+            lines[-1].add(token)
+
+    # combine all lines in each bracket block if that fits in max_inline_length
+    skip_lines = 0
+    for index, line in enumerate(lines):
+        if skip_lines > 0:
+            skip_lines -= 1
+            continue
+        yield from line.tokens
+        if line.can_start_inline_block:
+            indent_level = line.indent_level
+            line_length = indent_level + line.inline_length
+            pending_lines = 0
+            pending = []
+            last_token_is_opening_bracket = True
+            index = index + 1  # start on the next line
+            for line in lines[index:]:
+                if line.contains_comment:
+                    break
+                if not last_token_is_opening_bracket and not isinstance(
+                    line.inline_tokens[0], ClosingBracket
+                ):
+                    pending.append(Whitespace(" "))
+                    line_length += 1
+                pending_lines += 1
+                pending.extend(line.inline_tokens)
+                line_length += line.inline_length
+                last_token_is_opening_bracket = isinstance(
+                    line.inline_tokens[-1], OpeningBracket
+                )
+                if line_length > max_inline_length:
+                    break
+                if line.indent_level <= indent_level:
+                    # flush pending and handle next block if present
+                    yield from pending
+                    skip_lines += pending_lines
+                    if line.can_start_inline_block:
+                        pending_lines = 0
+                        pending = []
+                    else:
+                        break
+
+
+def reformat(query):
```

Comment:
If this module were to be put under test, a callback argument for the format type would make this a useful entrypoint.